docs/content/config/advanced/kubernetes.md

```diff
@@ -2,392 +2,374 @@
-    kind: Service
-    apiVersion: v1
+```yaml
+---
+apiVersion: apps/v1
+kind: Deployment
+
+metadata:
+  name: mailserver
+
+  annotations:
+    ignore-check.kube-linter.io/run-as-non-root: >-
+      The mailserver needs to run as root
+    ignore-check.kube-linter.io/privileged-ports: >-
+      The mailserver needs privilegdes ports
+    ignore-check.kube-linter.io/no-read-only-root-fs: >-
+      There are too many files written to make The
+      root FS read-only
+
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: mailserver
+
+  template:
     metadata:
-      name: mailserver
       labels:
         app: mailserver
+
+      annotations:
+        container.apparmor.security.beta.kubernetes.io/mailserver: runtime/default
+
     spec:
-      selector:
-        app: mailserver
-      ports:
-        - name: smtp
-          port: 25
-          targetPort: smtp
+      hostname: mailserver
+      containers:
+        - name: mailserver
+          image: ghcr.io/docker-mailserver/docker-mailserver:10.0.0
+          imagePullPolicy: IfNotPresent
+
+          securityContext:
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: false
+            runAsUser: 0
+            runAsGroup: 0
+            runAsNonRoot: false
+            privileged: false
+            capabilities:
+              add:
+                # file permission capabilities
+                - CHOWN
+                - FOWNER
+                - MKNOD
+                - SETGID
+                - SETUID
+                - DAC_OVERRIDE
+                # network capabilities
+                - NET_ADMIN
+                - NET_BIND_SERVICE
+                # miscellaneous  capabilities
+                - SYS_CHROOT
+                - SYS_PTRACE
+                - KILL
+              drop: [ALL]
+            seccompProfile:
+              type: RuntimeDefault
+
+          resources:
+            limits:
+              memory: 4Gi
+              cpu: 1500m
+            requests:
+              memory: 2Gi
+              cpu: 600m
+
+          volumeMounts:
+            - name: files
+              subPath: postfix-accounts.cf
+              mountPath: /tmp/docker-mailserver/postfix-accounts.cf
+              readOnly: true
+
+            # PVCs
+            - name: data
+              mountPath: /var/mail
+              subPath: data
+              readOnly: false
+            - name: data
+              mountPath: /var/mail-state
+              subPath: state
+              readOnly: false
+            - name: data
+              mountPath: /var/log/mail
+              subPath: log
+              readOnly: false
+
+            # other
+            - name: tmp-files
+              mountPath: /tmp
+              readOnly: false
+
+          ports:
+            - name: transfer
+              containerPort: 25
+              protocol: TCP
+            - name: esmtp-implicit
+              containerPort: 465
+              protocol: TCP
```

Comment:
This may be debatable, but 465 isn't a supported standard from what I'm aware, and nothing requires it (famous last words). Should we be good citizens and not advertise opening it up by removing it from here and the service?

Reply:
465 for implicit TLS is valid, it has a bit of a rough history being deprecated from STMPS (_port 25 with implicit TLS_) to being designated as implicit TLS `submissions` service (_as opposed to explicit `submission` service with StartTLS on port 587_).

That change requires auth like port 587, but plain TLS socket can be used, no StartTLS negotiation required. Due to the confusion from it's history, it's support is a bit spotty when offered by services.

587 had been around for a while and established itself, so is more common to see and without any confusion. That said StartTLS has also had some issues with how some software behaved (_eg IIRC.. sending auth credentials in plain-text if StartTLS failed to negotiate a secure exchange and the server rejected the client)).

Reply:
It's funny to see this debate everywhere :D @radicand you may want to read [this](https://github.com/authelia/authelia/issues/2194), and what @polarathene said is absolutely correct :)

TL;DR 465 is actually preferred over 578